src/troute-network/troute/HYFeaturesNetwork.py

```diff
@@ -431,64 +431,77 @@ def preprocess_data_assimilation(self, network):
+            lake_gage_hydroseq_df['lake_id'] = lake_gage_hydroseq_df['lake_id'].astype(int)
+            lake_gage_df = lake_gage_hydroseq_df[['lake_id','gages']].drop_duplicates()
+            lake_gage_hydroseq_df = lake_gage_hydroseq_df.groupby(['lake_id','gages']).max('hydroseq').reset_index().set_index('lake_id')
 
-            lake_gage_df = lake_gage_df[lake_gage_df['gages'].isin(lake_outflow_gage_ids)]
-            #####################################
             #FIXME: temporary solution, handles USGS and USACE reservoirs. Need to update for
             # RFC reservoirs...
+            #NOTE: In the event a lake ID has multiple gages, this also finds the gage furthest 
+            # downstream (based on hydroseq) separately for USGS and USACE crosswalks. 
             usgs_ind = lake_gage_df.gages.str.isnumeric()
             self._usgs_lake_gage_crosswalk = (
-                lake_gage_df.loc[usgs_ind].rename(columns={'lake_id': 'usgs_lake_id', 'gages': 'usgs_gage_id'})
-                .set_index('usgs_lake_id')
-                )
-            
+                lake_gage_df.loc[usgs_ind].rename(columns={'lake_id': 'usgs_lake_id', 'gages': 'usgs_gage_id'}).
+                set_index('usgs_lake_id').
+                merge(lake_gage_hydroseq_df.
+                      rename_axis('usgs_lake_id').
+                      rename(columns={'gages': 'usgs_gage_id'}), on=['usgs_lake_id','usgs_gage_id']).
+                sort_values(['usgs_gage_id','hydroseq']).groupby('usgs_lake_id').
+                last().
+                drop('hydroseq', axis=1)
+            )
+
             self._usace_lake_gage_crosswalk =  (
-                lake_gage_df.loc[~usgs_ind].rename(columns={'lake_id': 'usace_lake_id', 'gages': 'usace_gage_id'})
-                .set_index('usace_lake_id')
-                )
+                lake_gage_df.loc[~usgs_ind].rename(columns={'lake_id': 'usace_lake_id', 'gages': 'usace_gage_id'}).
+                set_index('usace_lake_id').
+                merge(lake_gage_hydroseq_df.
+                      rename_axis('usace_lake_id').
+                      rename(columns={'gages': 'usace_gage_id'}), on=['usace_lake_id','usace_gage_id']).
+                sort_values(['usace_gage_id','hydroseq']).groupby('usace_lake_id').
+                last().
+                drop('hydroseq', axis=1)
+            )
+            
+            # Set waterbody types if DA is turned on:
+            streamflow_nudging = self.data_assimilation_parameters.get('streamflow_da',{}).get('streamflow_nudging',False)
+            usgs_da = self.data_assimilation_parameters.get('reservoir_da',{}).get('reservoir_persistence_usgs',False)
+            usace_da = self.data_assimilation_parameters.get('reservoir_da',{}).get('reservoir_persistence_usace',False)
+            rfc_da = self.waterbody_parameters.get('rfc',{}).get('reservoir_rfc_forecasts',False)
```

Comment:
It seems like test_conus_AnA.yaml doesn't have reservoir_rfc_forecasts parameter. Instead, test_DA_config.yaml has it.  Don't we need to use self.data_assimilation_paramters instead then?

Reply:
Yeah, you're right. RFC parameters used to be under waterbody_parameters, but I moved it when working on the new troute.config module into data_assimilation_parameters where I think it makes more sense. I'll update this.